crowdfunding_compassion/models/sale_order_line.py

```diff
@@ -34,14 +34,17 @@ def get_donation_description(self, product):
         return super().get_donation_description(product)
 
     def _prepare_invoice_line(self, **optional_values):
-        # Propagate ambassador and event to invoice line
-        res = super()._prepare_invoice_line(**optional_values)
-        analytic = self.participant_id.project_id.event_id.analytic_id
-        res.update(
-            {
-                "user_id": self.participant_id.partner_id.id,
-                "analytic_account_id": analytic.id,
-                "crowdfunding_participant_id": self.participant_id.id,
-            }
-        )
+        res = super(SaleOrderLine, self)._prepare_invoice_line(**optional_values)
+
+        if hasattr(self, 'participant_id') and self.participant_id:
+            user_id = self.participant_id.partner_id.id
+            analytic_id = self.participant_id.project_id.event_id.analytic_id.id
+            crowdfunding_participant_id = self.participant_id.id,
+            res.update(
+                {
+                    "user_id": user_id,
+                    "analytic_account_id": analytic_id,
+                    "crowdfunding_participant_id": crowdfunding_participant_id,
+                }
+            )
```

Comment:
Cosmetic changes to reduce the number of lines.
```suggestion
        participant = getattr(self, 'participant_id', False)
        if participant:
            res.update(
                {
                    "user_id": participant.partner_id.id,
                    "analytic_account_id": participant.project_id.event_id.analytic_id.id,
                    "crowdfunding_participant_id": participant.id,
                }
            )
```